packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/index.ts

```diff
@@ -16,4 +16,23 @@ export { DiagramPaletteToolContext } from './renderer/palette/DiagramPaletteTool
 export type { DiagramPaletteToolContextValue } from './renderer/palette/DiagramPalette.types';
 export { DiagramPaletteToolContribution } from './renderer/palette/DiagramPaletteToolContribution';
 export type { DiagramPaletteToolContributionComponentProps } from './renderer/palette/DiagramPaletteToolContribution.types';
-export type { NodeData } from './renderer/DiagramRenderer.types';
+export type { Diagram, NodeData } from './renderer/DiagramRenderer.types';
+export type { NodeTypeComponentRegistry } from './contexts/NodeContext.types';
+export type { NodeStyleContribution } from './graphql/subscription/nodeFragment.types';
+export type { NodeTypeContextValue } from './contexts/NodeContext.types';
+export { NodeTypeContext } from './contexts/NodeContext';
+export type { ILayoutEngine, INodeLayoutHandler } from './renderer/layout/LayoutEngine.types';
+export type { DiagramNodeType } from './renderer/node/NodeTypes.types';
+export { GQLViewModifier } from './graphql/subscription/nodeFragment.types';
+export type { GQLNode, GQLNodeStyle } from './graphql/subscription/nodeFragment.types';
+export { BorderNodePositon } from './renderer/DiagramRenderer.types';
+export { convertLabelStyle } from './converter/ConvertEngine';
+export type { IConvertEngine, INodeConverterHandler } from './converter/ConvertEngine.types';
+export { NodeTypeContribution } from './renderer/node/NodeTypeContribution';
+export { useDrop } from './renderer/drop/useDrop';
+export { useConnector } from './renderer/connector/useConnector';
+export { useDropNode } from './renderer/dropNode/useDropNode';
+export { Label } from './renderer/Label';
+export { DiagramElementPalette } from './renderer/palette/DiagramElementPalette';
+export * from './renderer/layout/layoutNode';
+export * from './renderer/layout/layoutBorderNodes';
```

Comment:
I would have gone 100% the other direction by removing most of those. If every single tiny detail of all of our node implementation needs to be copied and pasted in every custom node we will break every single one of them during every releases. It's fine for a first version but we should warn consumers that we will break tons of those APIs and may even stop exporting any of them in the future (while still giving access to the same capabilities but in a more controlled manner) because we can very easily simplify a ton of copy / paste content in our node components.